easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -549,6 +549,21 @@ def _validate(self, attr, values):  # private method
         if self[attr] and self[attr] not in values:
             raise EasyBuildError("%s provided '%s' is not valid: %s", attr, self[attr], values)
 
+    def handle_external_module_metadata(self, dep_name):
+        """
+        helper function for _parse_dependency
+        handles metadata for external module dependencies
+        """
+        dependency = {}
+        if dep_name in self.external_modules_metadata:
+            dependency['external_module_metadata'].update(self.external_modules_metadata[dep_name])
```

Comment:
this can never work, since the `'external_module_metadata'` is not defined

also, you need to update what is there already